panther_manager/include/panther_manager/behavior_tree_utils.hpp

```diff
@@ -93,28 +96,76 @@ inline std::string GetLoggerPrefix(const std::string & bt_node_name)
 namespace BT
 {
 /**
- * @brief Converts a string to a vector of integers.
+ * @brief Converts a string to a vector of float.
  *
  * @param str The string to convert.
- * @return std::vector<int> The vector of integers.
+ * @return std::vector<float> The vector of float.
  *
- * @throw BT::RuntimeError Throws when there is no input or cannot parse int.
+ * @throw BT::RuntimeError Throws when there is no input or cannot parse float.
  */
 template <>
-inline std::vector<int> convertFromString<std::vector<int>>(StringView str)
+inline std::vector<float> convertFromString<std::vector<float>>(StringView str)
 {
-  auto parts = BT::splitString(str, ';');
-  if (!parts.size()) {
-    throw BT::RuntimeError("invalid input");
-  } else {
-    std::vector<int> result;
-    for (auto & part : parts) {
-      result.push_back(convertFromString<int>(part));
-    }
-
-    return result;
+  auto parts = splitString(str, ';');
+  std::vector<float> output;
+  output.reserve(parts.size());
+  for (const StringView & part : parts) {
+    output.push_back(convertFromString<float>(part));
   }
+  return output;
 }
+
+/**
+ * @brief Converts a string to a PoseStamped message.
+ *
+ * The string format should be "roll,pitch,yaw,x,y,z,frame_id" where:
+ *  - roll, pitch, yaw: Euler angles in radians.
```

Comment:
Are you sure that there are Euler angles? In the definition I see set `RPY`.

http://docs.ros.org/en/jade/api/tf2/html/classtf2_1_1Quaternion.html#a2ffba08e3fb4c10f9d5cbd6fc43399de

So the axes are X, Y, Z.

Reply:
There seems to be anything wrong. roll, pitch, yaw are Euler angle.

Reply:
It depends on the implementation of a order of axis rotation.


The best description will be from `tf2::Quaternion::setRPY()` what you used bellow. To clarify the description please use exact axis names.

```suggestion
 *  - roll, pitch, yaw: Rotation around axes XYZ.
```


 All of the angle are called roll, pitch, roll in functions:
 - [setRPY](https://docs.ros.org/en/humble/p/tf2/generated/classtf2_1_1Quaternion.html#_CPPv4N3tf210Quaternion6setRPYERK9tf2ScalarRK9tf2ScalarRK9tf2Scalar)
 - [setEuler](https://docs.ros.org/en/humble/p/tf2/generated/classtf2_1_1Quaternion.html#_CPPv4N3tf210Quaternion8setEulerERK9tf2ScalarRK9tf2ScalarRK9tf2Scalar)

Reply:
ok